pkg/objstore/gcs/gcs.go

```diff
@@ -47,8 +49,23 @@ func NewBucket(ctx context.Context, logger log.Logger, conf []byte, component st
 	if gc.Bucket == "" {
 		return nil, errors.New("missing Google Cloud Storage bucket name for stored blocks")
 	}
-	gcsOptions := option.WithUserAgent(fmt.Sprintf("thanos-%s/%s (%s)", component, version.Version, runtime.Version()))
-	gcsClient, err := storage.NewClient(ctx, gcsOptions)
+
+	var opts []option.ClientOption
+
+	// If ServiceAccount provided inside configuration use it, otherwise fallback to defaults
+	if gc.ServiceAccount != "" {
+		credentials, err := google.CredentialsFromJSON(ctx, []byte(gc.ServiceAccount))
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create credentials from JSON")
+		}
+		opts = append(opts, option.WithCredentials(credentials))
+	}
+
+	opts = append(opts,
```

Comment:
As a side thing (not a blocker for this PR), can we unsure, "debug" name from `debug.name`  flag is passed here? for component for example? It would help if use agent would be actually some name you configured I guess?